src/overlays/actors/ovl_Magic_Fire/z_magic_fire.c

```diff
@@ -15,7 +15,23 @@ void MagicFire_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void MagicFire_Update(Actor* thisx, GlobalContext* globalCtx);
 void MagicFire_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-/*
+void MagicFire_UpdateBeforeCast(Actor* thisx, GlobalContext* globalCtx);
+
+typedef enum {
+    /* 0x00 */ ACTION_INITIALIZE,
+    /* 0x01 */ ACTION_GROW_SLOWLY,
+    /* 0x02 */ ACTION_STOP_GROWING,
+    /* 0x03 */ ACTION_GROW_QUICKLY
+} MagicFireAction;
+
+typedef enum {
+    /* 0x00 */ SCREEN_TINT_NONE,
+    /* 0x01 */ SCREEN_TINT_FADE_IN,
+    /* 0x02 */ SCREEN_TINT_MAINTAIN,
```

Comment:
ideally both of these would have a prefix to make them specific to this system. even just DF_ for dins fire is fine